[http] Improve binding (#16282)
openhab/openhab-addons: bundles/org.openhab.binding.http/README.md

# HTTP Binding

> This binding allows using HTTP to bring external data into openHAB or execute HTTP requests on commands.  

Reviewer: ```suggestion
This binding allows using HTTP to bring external data into openHAB or execute HTTP requests on commands.
```

|-------------------------|----------|-------------|-----------------------------------------------------------------------------------------------------------|
| `stateExtension`        | yes      | -           | Appended to the `baseURL` for requesting states.                                                          |
| `commandExtension`      | yes      | -           | Appended to the `baseURL` for sending commands. If empty, same as `stateExtension`.                       |
| `stateTransformation  ` | yes      | -           | One or more transformation applied to received values before updating channel.                            |
| `commandTransformation` | yes      | -           | One or more transformation applied to channel value before sending to a remote.                           |
| `stateContent`          | yes      | -           | Content for state requests (if method is `PUT` or `POST`)                                                 |
| `mode`                  | no       | `READWRITE` | Mode this channel is allowed to operate. `READONLY` means receive state, `WRITEONLY` means send commands. |

> Transformations need to be specified in the same format as 

Reviewer: Did you mean to complete this sentence, since a space was added at the end?

Author: No. It should be deleted.

### `rollershutter`

| parameter   | optional | default | description                            |
|-------------|----------|---------|----------------------------------------|
| `upValue`   | yes      | -       | A special value that represents `UP`   |
| `downValue` | yes      | -       | A special value that represents `DOWN` |
| `stopValue` | yes      | -       | A special value that represents `STOP` |
| `moveValue` | yes      | -       | A special value that represents `MOVE` |

All values that are not `upValue`, `downValue`, `stopValue`, `moveValue` are interpreted as position 0-100% and need to be numeric only.
                    
### `switch`

| parameter  | optional | default | description                           |
|------------|----------|---------|---------------------------------------|
| `onValue`  | no       | -       | A special value that represents `ON`  |
| `offValue` | no       | -       | A special value that represents `OFF` |

**Note:** Special values need to be exact matches, i.e. no leading or trailing characters and comparison is case-sensitive.

## URL Formatting

> After concatenation of the `baseURL` and the `commandExtension` or the `stateExtension` (if provided) the URL is formatted using the [java.util.Formatter](http://docs.oracle.com/javase/6/docs/api/java/util/Formatter.html).

Reviewer: In the previous version of the PR this was changed to:
```suggestion
After concatenation of the `baseURL` and the `commandExtension` or the `stateExtension` (if provided) the URL is formatted using the [java.util.Formatter](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Formatter.html).
```

See the link above for more information about the available format parameters (e.g. to use the string representation, you need to append `s` to the reference, for a timestamp `t`).
When sending an OFF command on 2020-07-06, the URL

```
http://www.domain.org/home/lights/23871/?status=%2$s&date=%1$tY-%1$tm-%1$td
``` 

> is transformed to 

Reviewer: ```suggestion
is transformed to
```

See the link above for more information about the available format parameters (e.g. to use the string representation, you need to append `s` to the reference, for a timestamp `t`).
When sending an OFF command on 2020-07-06, the URL

```
http://www.domain.org/home/lights/23871/?status=%2$s&date=%1$tY-%1$tm-%1$td
``` 

is transformed to 

```
http://www.domain.org/home/lights/23871/?status=OFF&date=2020-07-06
```

## Examples

### `demo.things`

> ```

Reviewer: ```suggestion
```java
```

can you also change the indentation back to four spaces rather than tab?
